nc_time_axis/tests/unit/test_NetCDFTimeDateLocator.py

```diff
@@ -115,5 +115,43 @@ def test_yearly(self):
             self.check(5, 0, 5*365), [31., 485., 942., 1399., 1856.])
 
 
+class Test_tick_values_yr0(unittest.TestCase):
+    def setUp(self):
+        self.date_unit = 'days since 0001-01-01 00:00'
+
+    def check(self, max_n_ticks, num1, num2, calendar):
+        locator = NetCDFTimeDateLocator(max_n_ticks=max_n_ticks,
+                                        calendar=calendar,
+                                        date_unit=self.date_unit)
+        return locator.tick_values(num1, num2)
+
+    def test_yearly_yr0_remove(self):
+            yr0_remove_calendars = [
+                "proleptic_gregorian",
+                "gregorian",
+                "julian",
+                "standard"
+                    ]
+
+            for calendar in [
+                'standard',
+                'gregorian',
+                'proleptic_gregorian',
+                'noleap',
+                '365_day',
+                '360_day',
+                'julian',
+                'all_leap',
+                '366_day',
+                    ]:
+
+                # convert values to dates, check that none of them has year 0
+                ticks = self.check(5, 0, 100*365, calendar)
+                if calendar in yr0_remove_calendars:
+                    for t in ticks:
+                        assert cftime.utime(
+                            self.date_unit,
+                            calendar).num2date(t).year != 0
```

Comment:
We should be able to get this a bit neater, without a loop, by using assignment and list comprehension.  List comprehension is usually more efficient, and in my opinion the below is a little more readable too. What do you think of this?

(I've written this as a pure re-write, not accounting for my other suggestions)

```suggestion
num2date = cftime.utime(self.date_unit, calendar).num2date
year_ticks = [num2date(t).year for t in ticks]
if calendar in yr0_remove_calendars:
    self.assertNotIn(0, year_ticks)
```

Reply:
Yes that is great. The initial one was kind of a mess, since I am not very familiar with the internal structure of the package. Thanks a lot for the suggestions.